browser/ui/side_panel/brave_contextual_side_panel_tab_helper.cc

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/side_panel/brave_contextual_side_panel_tab_helper.h"
+
+#include "brave/browser/ui/side_panel/brave_side_panel_utils.h"
+
+BraveContextualSidePanelTabHelper::BraveContextualSidePanelTabHelper(
+    content::WebContents* contents)
+    : WebContentsUserData(*contents) {
+  brave::RegisterContextualSidePanel(contents);
```

Comment:
Optional: does this code still need to live in `brave_side_panel_utils`? We could probably just move it into this constructor as it isn't used anywhere else.

Reply:
Its implementation is view specific so defined at `browser/ui/views/side_panel`.
Only decl is in `browser/ui/side_panel` to call it from outside of view layer.
So, decl and def are in separated layer.